mm2src/gossipsub/src/behaviour.rs

```diff
@@ -1290,11 +1308,35 @@ impl NetworkBehaviour for Gossipsub {
         self.peer_topics.insert(*id, Vec::new());
     }
 
-    fn inject_disconnected(&mut self, id: &PeerId) {
+    fn inject_connection_closed(
+        &mut self,
+        peer_id: &PeerId,
+        disconnected_conn_id: &ConnectionId,
+        disconnected_point: &ConnectedPoint,
+        _: <Self::ConnectionHandler as IntoConnectionHandler>::Handler,
+        remaining_established: usize,
+    ) {
+        let mut clean = false;
+
+        if let Some(connected_points) = self.peer_connections.get_mut(peer_id) {
+            connected_points.retain(|(conn_id, _point)| conn_id != disconnected_conn_id);
+            clean = connected_points.is_empty();
+        }
+        if clean {
+            self.peer_connections.remove(peer_id);
+        }
```

Comment:
This can be rewritten as follows without using the extra `clean` var
```
        if let Entry::Occupied(mut o) = self.peer_connections.entry(*peer_id) {
            let connected_points = o.get_mut();
            connected_points.retain(|(conn_id, _point)| conn_id != disconnected_conn_id);
            if connected_points.is_empty() {
                o.remove_entry();
            }
        }
```